pkg/versionwriter/version_writer.go

```diff
@@ -0,0 +1,63 @@
+package versionwriter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"aspect.build/cli/buildinfo"
+)
+
+type Format int
+
+const (
+	GNU Format = iota
+	Conventional
+
+	NotCleanVersionSuffix = " (with local changes)"
+	NoReleaseVersion      = "unknown [not built with --stamp]"
+)
+
+type VersionWriter struct {
```

Comment:
feels to me like this whole package is just a fancy "toString()" for a BuildInfo - maybe these functions could live there instead?

Reply:
I struggled with this. I ended up with a separate package because we were carrying around the `Format` state that is coming from a flag. Let me see what the code looks like if I move it to `buildinfo.BuildInfo`.

Reply:
I moved the logic to `buildinfo`.